modules/navier_stokes/src/variables/INSFVVariable.C

```diff
@@ -29,6 +30,62 @@ INSFVVariable::INSFVVariable(const InputParameters & params)
 {
 }
 
+void
+INSFVVariable::initialSetup()
+{
+  MooseVariableFVReal::initialSetup();
+
+  _boundary_id_to_separator.clear();
+  std::vector<FVFluxBC *> bcs;
+
+  const auto base_query = this->_subproblem.getMooseApp()
+                              .theWarehouse()
+                              .query()
+                              .template condition<AttribSystem>("FVFluxBC")
+                              .template condition<AttribThread>(_tid);
+
+  for (const auto bnd_id : this->_mesh.getBoundaryIDs())
+  {
+    auto base_query_copy = base_query;
+    base_query_copy.template condition<AttribBoundaries>(std::set<BoundaryID>({bnd_id}))
+        .queryInto(bcs);
+    for (const auto bc : bcs)
+    {
+      const auto separator = dynamic_cast<const INSFVHydraulicSeparatorInterface *>(bc);
+      if (separator)
+        _boundary_id_to_separator.emplace(bnd_id, bc);
```

Comment:
Yea you already have the cast here, so let's make to value type for this map `INSFVHydraulicSeparatorInterface`